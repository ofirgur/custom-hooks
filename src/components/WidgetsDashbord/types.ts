export type WidgetDashboardItem = {
  key: string;
  name: string;
  path: string;
  title: string;
  icon: string;
};
