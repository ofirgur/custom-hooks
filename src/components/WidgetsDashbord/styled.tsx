import styled from '@emotion/styled';

export const Styled = styled.div`
  min-height: 100vh;
  padding: 24px;
  background-color: #16191D;
`;

export const Title = styled.h2`
  text-align: center;
  height: 70px;
  color: #fff;
`;

export const Widgets = styled.div`
  display: flex;
  flex-direction: column;
  gap: 24px;
  @media only screen and (min-width: 690px) {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
`;
