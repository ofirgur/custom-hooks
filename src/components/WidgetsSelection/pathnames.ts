export const PATHNAMES = {
  home: '/',
  tarotMemoryGame: 'tarotmemorygame',
  tarotQuizGame: 'tarotquizgame',
};
