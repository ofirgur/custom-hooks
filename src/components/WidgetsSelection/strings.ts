export const STRINGS = {
  TAROT_MEMORY_GAME: 'Tarot Memory Game',
  TAROT_QUIZ_GAME: 'Tarot Quiz Game',
  BOOKS_GAME: 'Books Game',
  PRAYERS_GENERATOR: 'Prayer Generator',
};
