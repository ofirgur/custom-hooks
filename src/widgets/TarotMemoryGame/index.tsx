import React, { useEffect } from 'react';
import { INITIAL_GAME_STATE, GAME_STATE_LOCAL_STORAGE_NAME } from './utils/constants';
import { getGameStateLocalStorage, removeGameStateLocalStorage } from '../../localstorage';
import GameOutletScreen from './components/GameOutletScreen';
import { GameStateContextProvider } from './utils/context';

const TarotMemoryGame = () => {
  const GAME_STATE_LOCAL_STORAGE = getGameStateLocalStorage(GAME_STATE_LOCAL_STORAGE_NAME);
  useEffect(() => {
    removeGameStateLocalStorage(GAME_STATE_LOCAL_STORAGE_NAME);
  }, []);

  return (
    <GameStateContextProvider initialState={GAME_STATE_LOCAL_STORAGE || INITIAL_GAME_STATE}>
      <GameOutletScreen />
    </GameStateContextProvider>
  );
};

export default TarotMemoryGame;
