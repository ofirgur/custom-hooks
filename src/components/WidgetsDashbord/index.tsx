import React from 'react';
import { WidgetDashboardItem } from './types';
import LinkCard from './components/LinkCard';
import { Styled, Title, Widgets } from './styled';

type WidgetsDashbordProps = {
  widgets: WidgetDashboardItem[];
};

const WidgetsDashbord = (props: WidgetsDashbordProps) => {
  const { widgets } = props;

  return (
    <Styled>
      <Title>EMAIL ACTIVITY WIDGETS</Title>
      <Widgets>
        {
          widgets.map((widget, index) => (
            <LinkCard key={index} widget={widget} />
          ))
        }
      </Widgets>
    </Styled>
  );
};

export default WidgetsDashbord;
