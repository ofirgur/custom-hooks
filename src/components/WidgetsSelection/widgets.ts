import { PATHNAMES } from './pathnames';
import { STRINGS } from './strings';

export const WIDGETS = [
  {
    key: 'TarotMemoryGame',
    name: 'tarotmemorygame',
    path: PATHNAMES.tarotMemoryGame,
    title: STRINGS.TAROT_MEMORY_GAME,
    icon: '',
  },
  {
    key: 'TarotQuizGame',
    name: 'tarotquizgame',
    path: PATHNAMES.tarotQuizGame,
    title: STRINGS.TAROT_QUIZ_GAME,
    icon: '',
  },
];
