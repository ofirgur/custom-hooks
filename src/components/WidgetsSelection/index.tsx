import React, { Suspense } from 'react';
import { BrowserRouter, Outlet, Route, Routes } from 'react-router-dom';
import WidgetsDashbord from '../WidgetsDashbord';
import { PATHNAMES } from './pathnames';
import { WIDGETS } from './widgets';

const TarotMemoryGame = React.lazy(() => import(/* webpackChunkName: "tarot-memory-game" */'../../widgets/TarotMemoryGame'));
const TarotQuizGame = React.lazy(() => import(/* webpackChunkName: "tarot-quiz-game" */'../../widgets/TarotQuizGame'));
const TestAndStyleGame = React.lazy(() => import(/* webpackChunkName: "test-and-style-game" */'../../widgets/TestAndStyleGame'));

const games = {
  [PATHNAMES.tarotMemoryGame]: TarotMemoryGame,
  [PATHNAMES.tarotQuizGame]: TarotQuizGame,
};

const WidgetsSelection = ({ widgetName }: { widgetName: string }) => {
  if (games[widgetName]) {
    const Widget = games[widgetName];
    return (
      <Suspense fallback={<>...</>}>
        <Widget />
      </Suspense>
    );
  }
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Outlet />}>
          <Route index element={<WidgetsDashbord widgets={WIDGETS} />} />
          <Route
            path={PATHNAMES.tarotMemoryGame}
            element={(
              <Suspense fallback={<>...</>}>
                <TarotMemoryGame />
              </Suspense>
            )}
          />
          <Route
            path={PATHNAMES.tarotQuizGame}
            element={(
              <Suspense fallback={<>...</>}>
                <TarotQuizGame />
              </Suspense>
            )}
          />
          <Route
            path={PATHNAMES.testandstylegame}
            element={(
              <Suspense fallback={<>...</>}>
                <TestAndStyleGame />
              </Suspense>
            )}
          />
        </Route>
      </Routes>
    </BrowserRouter>
  );
};

export default WidgetsSelection;
