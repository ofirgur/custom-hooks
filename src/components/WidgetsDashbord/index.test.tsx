import * as React from 'react';
import { BrowserRouter, Outlet, Route, Routes } from "react-router-dom";
import { render } from '@testing-library/react';

import { WIDGETS } from '../WidgetsSelection/widgets';
import WidgetsDashbord from './';

const component = (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<Outlet />}>
        <Route index element={<WidgetsDashbord widgets={WIDGETS} />} />
      </Route>
    </Routes>
  </BrowserRouter>
);

describe('WidgetsDashboard', () => {
  it('should render correctly', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
