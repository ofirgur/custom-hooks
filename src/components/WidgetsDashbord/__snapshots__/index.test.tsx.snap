// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetsDashboard should render correctly 1`] = `
<div>
  <div
    class="css-8gwbof"
  >
    <h2
      class="css-7c3lnj"
    >
      EMAIL ACTIVITY WIDGETS
    </h2>
    <div
      class="css-rm55dl"
    >
      <div
        class="css-1flu1tv"
      >
        Tarot Memory Game
      </div>
      <div
        class="css-1flu1tv"
      >
        Tarot Quiz Game
      </div>
      <div
        class="css-1flu1tv"
      >
        Books Game
      </div>
      <div
        class="css-1flu1tv"
      >
        Prayer Generator
      </div>
    </div>
  </div>
</div>
`;
